src/routes/settings-usage-consolidation-route.test.js

```diff
@@ -0,0 +1,293 @@
+}));
+
+const currencies = {
+  isLoading: false,
+  items: [{
+    attributes: {
+      code: 'AFN',
+      description: 'Afghan Afghani',
+    },
+  }, {
+    attributes: {
+      code: 'ALL',
+      description: 'Albanian Lek',
+    },
+  }],
+  errors: [],
+};
+const ucCredentials = {
+  isPresent: false,
+  isLoading: false,
+  isFailed: false,
+  isUpdated: false,
+  errors: [],
+};
+const usageConsolidation = {
+  isLoading: false,
+  isLoaded: false,
+  isFailed: false,
+  isKeyLoading: false,
+  isKeyLoaded: false,
+  isKeyFailed: false,
+  hasSaved: false,
+  data: {},
+  errors: [],
+};
+
+const renderSettingsUsageConsolidationRoute = ({ harnessProps = {}, props = {} }) => render(
+  <MemoryRouter>
+    <Harness
+      storeInitialState={{
+        data: {
+          currencies,
+          ucCredentials,
+          usageConsolidation,
+        },
+      }}
+      {...harnessProps}
+    >
+      <SettingsUsageConsolidationRoute
+        clearUsageConsolidationErrors={noop}
+        getUsageConsolidation={noop}
+        getUsageConsolidationKey={noop}
+        patchUsageConsolidation={noop}
+        postUsageConsolidation={noop}
+        getCurrencies={noop}
+        getUcCredentials={noop}
+        updateUcCredentials={noop}
+        {...props}
+      />
+    </Harness>
+  </MemoryRouter>
+);
+
+describe('Given SettingsUsageConsolidationRoute', () => {
+  beforeEach(() => {
+    mockGetUsageConsolidation.mockClear();
+    mockGetUsageConsolidationKey.mockClear();
+    mockPatchUsageConsolidation.mockClear();
+    mockPostUsageConsolidation.mockClear();
+    mockGetCurrencies.mockClear();
+    mockGetUcCredentials.mockClear();
+    mockUpdateUcCredentials.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should handle getCurrencies', async () => {
+    await act(async () => {
+      await renderSettingsUsageConsolidationRoute({
+        props: { getCurrencies: mockGetCurrencies },
+      });
+    });
+
+    expect(mockGetCurrencies).toHaveBeenCalled();
+  });
+
+  it('should handle getUcCredentials', async () => {
+    await act(async () => {
+      await renderSettingsUsageConsolidationRoute({
+        props: { getUcCredentials: mockGetUcCredentials },
+      });
+    });
+
+    expect(mockGetUcCredentials).toHaveBeenCalled();
+  });
+
+  describe('when usage consolidation is loading', () => {
+    it('should show spinner', () => {
+      const { container } = renderSettingsUsageConsolidationRoute({
+        harnessProps: {
+          storeInitialState: {
+            data: {
+              currencies,
+              ucCredentials,
+              usageConsolidation: {
+                ...usageConsolidation,
+                isLoading: true,
+              },
+            },
+          },
+        },
+      });
+
+      expect(container.querySelector('.icon-spinner-ellipsis')).toBeDefined();
+    });
+  });
+
+  describe('when usage consolidation is loaded', () => {
+    it('should handle getUsageConsolidationKey', async () => {
```

Comment:
Previous and this test have same props, but are supposed to test different cases? Am I missing something?

Reply:
In first test we use prop `isLoading` to see is spinner displayed and in the second we use `isLoaded` prop, when data loaded and we handle other action

Reply:
In first test we use prop `isLoading` to see is spinner displayed and in the second we use `isLoaded` prop, when data loaded and we handle other action

Reply:
You're right, sorry, I didn't read that correctly